### ISSUE TITLE
[Windows] Why the initial project call Destroy inside Win32Window::CreateAndShow, that's not make sense

### ISSUE THREAD

**Issue by ShaowenSu** [flutter/flutter#107780]
1. Run `flutter create test`
2. Check the `win32_window.cpp` under `runner` folder
3. Check the `CreateAndShow()` method
4. It calls `Destroy()` at the beginning and that makes `OnDestroy()` running. It does not meet the semantic.

```
[✓] Flutter (Channel stable, 2.10.5, on macOS 12.4 21F79 darwin-x64, locale en-CN)
    • Flutter version 2.10.5 at /Users/shaowensu/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5464c5bac7 (3 months ago), 2022-04-18 09:55:37 -0700
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2
```

**Comment by huycozy** [comment]
Hi @ShaowenSu, thanks for filing the issue.
On master, the `Destroy()` is called on `CreateAndShow` method: 
https://github.com/flutter/flutter/blob/2f0a25257e485e724f0daa4af99e63fd04a8fc5d/packages/flutter_tools/templates/app_shared/windows.tmpl/runner/win32_window.cpp#L105-L109

This looks like a proposal. Could you please describe what you would like Flutter to be able to do?

**Comment by ShaowenSu** [comment]
> Hi @ShaowenSu, thanks for filing the issue. On master, the `Destroy()` is called on `CreateAndShow` method:
> 
> https://github.com/flutter/flutter/blob/2f0a25257e485e724f0daa4af99e63fd04a8fc5d/packages/flutter_tools/templates/app_shared/windows.tmpl/runner/win32_window.cpp#L105-L109
> 
> This looks like a proposal. Could you please describe what you would like Flutter to be able to do?

Hi @huycozy, I am thinking, calling `Destroy()` on `CreateAndShow` is not making any sense. It doesn't  meet the semantic. For example, I would put something inside `OnDestroy()` such like Closing the file, Disconnecting the network, Cleaning the systray, etc., but since `Destroy()` is called on `CreateAndShow`, which would be called every time the application is launching, then `OnDestroy()` will be called at the same time, this mess up everything.

What do you thing?

**Comment by huycozy** [comment]
@ShaowenSu thanks for your update. Labeling the issue for further opinion from the team.

**Comment by gspencergoog** [comment]
This is definitely a valid criticism.  We should see what happens if we just don't call `Destroy`, and if necessary, refactor `OnDestroy` into `OnCreate` and `OnDestroy`.

**Comment by ShaowenSu** [comment]
> This is definitely a valid criticism. We should see what happens if we just don't call `Destroy`, and if necessary, refactor `OnDestroy` into `OnCreate` and `OnDestroy`.

Yes, I have tried to remove the `Destroy()` call from `CreateAndShow()`. Nothing happens.
